x-pack/metricbeat/module/googlecloud/pubsub/_meta/fields.yml

```diff
@@ -0,0 +1,70 @@
+- name: pubsub
+  release: beta
+  type: group
+  description: Google Cloud PubSub metrics
+  fields:
+    - name: subscription
+      type: group
+      description: Suscription related metrics
+      fields:
+        - name: ack_message_count
+          type: long
+          description: Cumulative count of messages acknowledged by Acknowledge requests, grouped by delivery type.
+        - name: backlog_bytes
+          type: long
+          description: Total byte size of the unacknowledged messages (a.k.a. backlog messages) in a subscription.
+        - name: num_outstanding_messages
+          type: long
+          description: Number of messages delivered to a subscription's push endpoint, but not yet acknowledged.
+        - name: num_undelivered_messages
+          type: long
+          description: Number of unacknowledged messages (a.k.a. backlog messages) in a subscription.
+        - name: oldest_unacked_message_age
+          type: long
+          description: Age (in seconds) of the oldest unacknowledged message (a.k.a. backlog message) in a subscription.
+        - name: pull_ack_message_operation_count
+          type: long
+          description: Cumulative count of acknowledge message operations, grouped by result. For a definition of message operations, see Cloud Pub/Sub metric subscription/mod_ack_deadline_message_operation_count.
+        - name: pull_ack_request_count
+          type: long
+          description: Cumulative count of acknowledge requests, grouped by result.
+        - name: pull_message_operation_count
+          type: long
+          description: Cumulative count of pull message operations, grouped by result. For a definition of message operations, see Cloud Pub/Sub metric subscription/mod_ack_deadline_message_operation_count.
+        - name: pull_request_count
```

Comment:
do you think merging some of these fields will be better? For example, pull_request_count and push_request_count can be request_count.pull and request_count.push.

Reply:
Definitely but, because we are using lightweight modules, no field naming modification "is possible" (or, at least, not without making code way more complex now)